### ISSUE TITLE
Consolidate Vector & Array

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#5011]
This task is automatically imported from the old Task Issue Board and it was originally created by jaroslavtulach.
Original issue is [here](https://www.pivotaltracker.com/story/show/183217534).
<hr />

#183000876 has removed the difference between Enso `Array` & `Vector` - they both _"duck like a polyglot array"_. The only difference is that `Vector` originates from _Enso_ and is _idealized_ as _immutable_. `Array` comes from Java, JavaScript, Python, etc. and nobody knows much about it.

The next step is to consolidate them more:
- methods available on `Vector` and `Array` shall be the same (as much as possible)
- methods needlessly converting between them can be removed - like `Vector.to_array`, `Vector.unsafe_at`
- the `Vector.from_polyglot_array` method shall be renamed - see [discussion](https://github.com/enso-org/enso/pull/3667#discussion_r964804325)

## Tasks:
- [x] Write a test that compares methods available on `Vector` and `Array` type
- [x] Make sure the same methods are available in both types
- [x] Make sure `Vector` and _polyglot array_ can be used interchangeably when calling builtins

## Deferred Tasks

- [ ] Remove methods that make no sense anymore (likely `Vector.to_array`)
- [ ] Find better name for `Vector.from_polyglot_array`
- [ ] Benchmark using `at` instead of `unsafe_at` and if fine, remove `unsafe_at`
- [ ] Remove needless conversions to `Vector` - like [(Vector.from_array arr) . map](https://github.com/enso-org/enso/pull/3667/files#diff-7e2fdad832e7ad6916e8873a78bbc3129d96f9c813230c7240af93193e3f3913R23)

## Blockers:
[#183000876](https://www.pivotaltracker.com/story/show/#183000876) resolved
[#183266964](https://www.pivotaltracker.com/story/show/#183266964) resolved
[#183058716](https://www.pivotaltracker.com/story/show/#183058716) resolved
[#184278959](https://www.pivotaltracker.com/story/show/#184278959) resolved

## Comments:
To _consolidate Vector & Array_ we want to make sure both `Vector` and `Array` can be used interchangeably. That's not currently the case. There are two known problems where explicit convertsion `to_array` is still required:
* #183266964
* #183058716

I am afraid we have to address these two sample _needless conversions in builtins_ before we can claim a victory in _consolidating Vector & Array_.
 (jaroslavtulach - Sep 29, 2022)
<hr />
**Hubert Plociniczak** reports a new **STANDUP** for yesterday (2023-01-09):

**Progress:** QA, trying to work on consolidation but little work done due to other distractions. Investigated double reporting of errors as noticed by Radek (will report the ticket once I have something tangible). It should be finished by 2023-01-12.

**Next Day:** Next day I will be working on the [#183217534 task](https://www.pivotaltracker.com/story/show/183217534). More QA and consolidation work. (Enso Bot - Jan 10, 2023)
<hr />
**Hubert Plociniczak** reports a new **STANDUP** for yesterday (2023-01-11):

**Progress:** QA, problems with Windows setup to reproduce vcs issues. Minor tweaks to consolidation work (decided to postpone until James' work on API improvements lands. It should be finished by 2023-01-12.

**Next Day:** Next day I will be working on the [#183217534 task](https://www.pivotaltracker.com/story/show/183217534). Look into resolving fully qualified names, pick up vector/array work. (Enso Bot - Jan 12, 2023)
<hr />
